testsupport/init.go

```diff
@@ -40,48 +52,93 @@ func WaitForDeployments(t *testing.T, obj runtime.Object) (*framework.Context, *
-	require.NoError(t, err, "failed while waiting for member operator deployment")
-
 	// wait for registration service to be ready
-	err = e2eutil.WaitForDeployment(t, f.KubeClient, registrationServiceNs, "registration-service", 2, wait.DefaultRetryInterval, wait.DefaultOperatorTimeout)
+	err = sdkutil.WaitForDeployment(t, f.KubeClient, registrationServiceNs, "registration-service", 2, wait.DefaultRetryInterval, wait.DefaultOperatorTimeout)
 	require.NoError(t, err, "failed while waiting for registration service deployment")
 
-	awaitility := &wait.Awaitility{
-		T:                     t,
-		Client:                f.Client,
-		KubeClient:            f.KubeClient,
-		ControllerClient:      f.Client.Client,
-		Scheme:                f.Scheme,
-		HostNs:                hostNs,
-		MemberNs:              memberNs,
-		RegistrationServiceNs: registrationServiceNs,
-	}
-
-	registrationServiceRoute, err := awaitility.Host().WaitForRouteToBeAvailable(registrationServiceNs, "registration-service", "/")
+	registrationServiceRoute, err := hostAwait.WaitForRouteToBeAvailable(registrationServiceNs, "registration-service", "/")
 	require.NoError(t, err, "failed while waiting for registration service deployment")
 
 	registrationServiceURL := "http://" + registrationServiceRoute.Spec.Host
 	if registrationServiceRoute.Spec.TLS != nil {
 		registrationServiceURL = "https://" + registrationServiceRoute.Spec.Host
 	}
-	awaitility.RegistrationServiceURL = registrationServiceURL
-	err = awaitility.WaitForReadyToolchainClusters()
+	hostAwait.RegistrationServiceURL = registrationServiceURL
+
+	// wait for member operator to be ready
+	memberCluster, err := hostAwait.WaitForToolchainClusterWithCondition(cluster.Member, memberNs, wait.ReadyToolchainCluster)
+	require.NoError(t, err)
+	memberConfig, err := buildConfig(f.Client.Client, memberCluster)
+
+	kubeClient, err := kubernetes.NewForConfig(memberConfig)
+	require.NoError(t, err)
+	err = sdkutil.WaitForDeployment(t, kubeClient, memberNs, "member-operator", 1, wait.DefaultOperatorRetryInterval, wait.DefaultOperatorTimeout)
+	require.NoError(t, err, "failed while waiting for member operator deployment")
+
+	memberClient, err := client.New(memberConfig, client.Options{
+		Scheme: f.Scheme,
+	})
+	require.NoError(t, err)
+	memberAwait := wait.NewMemberAwaitility(t, memberClient, memberNs)
+
+	_, err = memberAwait.WaitForToolchainClusterWithCondition(hostAwait.Type, hostAwait.Namespace, wait.ReadyToolchainCluster)
 	require.NoError(t, err)
 
 	t.Log("both operators are ready and in running state")
-	return ctx, awaitility
+	return ctx, hostAwait, memberAwait
+}
+
+const (
+	toolchainAPIQPS   = 20.0
+	toolchainAPIBurst = 30
+	toolchainTokenKey = "token"
+)
+
+// TODO: reuse code in https://github.com/codeready-toolchain/toolchain-common/blob/master/pkg/cluster/service.go#L137 instead
```

Comment:
I was thinking about addressing this comment in a subsequent PR, this one being already big enough IMO

Reply:
ok, as you wish :wink: I was asking because this change would make the PR actually smaller...

Reply:
yeah, maybe 🤔 To be honest, I left this note as a reminder because at the time I was deep in the refactoring, with many other parts to take care of (and which I wanted to focus on, first). Anyways, let me address this in a subsequent PR for now.